lib/hammer_cli/defaults.rb

```diff
@@ -0,0 +1,139 @@
+require 'hammer_cli'
+require 'yaml'
+module HammerCLI
+
+  class DefaultsCommand < HammerCLI::AbstractCommand
+
+    class ListDefaultsCommand < HammerCLI::DefaultsCommand
+      command_name 'list'
+
+      desc _('List all the defaults parameters')
+      def execute
+        list_all_defaults_message
+        HammerCLI::EX_OK
+      end
+    end
+
+    class AddDefaultsCommand < HammerCLI::DefaultsCommand
+      command_name 'add'
+
+      desc _('Add a default parameter to config')
+      option "--param-name", "Option_name", _("The name of the default option"), :required => true
+      option "--param-val", "Option_value", _("The value for the default option")
+      option "--plugin-name", "Option_name", _("The plugin name defaults will be generated for")
+
+      def execute
+        begin
+          if option_plugin_name.nil? && option_param_val.nil? || !option_plugin_name.nil? && !option_param_val.nil?
+            bad_input
+          else
+            namespace = ("HammerCLI" + option_plugin_name.split('_').collect!{ |w| w.downcase.capitalize }.join + "::Defaults") if option_plugin_name
+            if option_plugin_name
+              raise NameError unless HammerCLI::Defaults.providers.any? {|p| p.to_s.include? namespace}
+              raise NotImplementedError unless HammerCLI::Defaults.providers[0].support? option_param_name
+            end
+            Defaults.add_defaults_to_conf({option_param_name => option_param_val},option_plugin_name ? HammerCLI::Defaults.providers.find_index(namespace.constantize) : "")
```

Comment:
I understand the `0` in the config file now. Storing the index may break when you add a plugin later and the alphabetical order changes. It would be better to store the name directly.